x/programs/test/programs/context_injection/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+use wasmlanche_sdk::{public, state_keys, Context};
+
+#[state_keys]
+pub enum StateKeys {
+    LastTimestamp,
+    LastHeight,
+    LastActor,
+}
+
+#[public]
+pub fn can_change_timestamp(context: Context<StateKeys>) {
+    let new_timestamp = context.timestamp;
+    match context
+        .program
+        .state()
+        .get(StateKeys::LastTimestamp)
+        .unwrap()
+    {
+        Some(timestamp) => {
+            assert!(new_timestamp > timestamp)
+        }
+        None => {
+            context
+                .program
+                .state()
+                .store(StateKeys::LastTimestamp, &new_timestamp)
+                .unwrap();
+        }
+    }
```

Comment:
Why don't you just return the timestamp then validate that it was the one that was injected. This is overcomplicated.

Reply:
true... Your idea is more "atomic"